src/components/map/index.tsx

```diff
@@ -37,17 +37,20 @@ export type {
 } from './use-map-events';
 
 export type MapCameraProps = {
-  center: google.maps.LatLngLiteral;
-  zoom: number;
   heading?: number;
   tilt?: number;
-};
+} & (
+  | {center: google.maps.LatLngLiteral}
+  | {defaultCenter: google.maps.LatLngLiteral}
+) &
+  ({zoom: number} | {defaultZoom: number});
 
 /**
  * Props for the Google Maps Map Component
  */
 export type MapProps = google.maps.MapOptions &
   MapEventProps &
+  MapCameraProps &
```

Comment:
`zoom` and `center` are defined in `google.maps.MapOptions` so they haven't been repeated here.

`MapCameraProps` isn't included since it is intended to be used together with the `MapCameraChangedEvent` (it's a type you can use for the details of that event), see here for example:
https://github.com/visgl/react-google-maps/blob/9618024ac9dff76b03b0cb6758e974b0361de1f0/examples/multiple-maps/src/app.tsx#L33-L41

Maybe it should be renamed to `MapCameraState` to avoid confusion with the props?

Reply:
Thanks. I would add that rename.